lib/CL/pocl_llvm_api.cc

```diff
@@ -1062,7 +1062,8 @@ static PassManager& kernel_compiler_passes
   passes.push_back("mem2reg");
   passes.push_back("domtree");
   passes.push_back("break-constgeps");
-  passes.push_back("automatic-locals");
+  if(!device->spmd)
```

Comment:
Let's add another parameter here as this might not be true for all future SPMD devices. For example "autolocals_to_args" or similar which will be FALSE for HSA devices, true by default.